### PR TITLE
Touch up types

### PR DESCRIPTION
**Pull request by rekmarks**
- Add full docstrings to declarations file
  - The docstrings in the JavaScript source are not picked up by TypeScript, at least in VS Code.
  - Make minor edits to source docstrings in process.
- Add typings for all `send` method signatures
- Use `RequestArguments` interface for `request` method
- Require `opts.connectionStream` param for `initProvider`
- Update `jsonrpc` property on `JsonRpcRequest` and `JsonRpcResponse` interfaces
- Update `JsonRpcResponse` interface to type with either `result` or `error` property
- Lint types (per our config, but without adding dependencies)
  - Basically adds spacing and semicolons

### CODE REVIEW

**Review by brad-decker — APPROVED** [review]

File: index.d.ts [hunk 1]
```diff
@@ -1,70 +1,174 @@
...
+  /**
+   * The user's currently selected Ethereum address.
+   * If null, MetaMask is either locked or the user has not permitted any
+   * addresses to be viewed.
+   */
+  readonly selectedAddress: string | null;
+
+  /**
+   * The network ID of the currently connected Ethereum chain.
+   * @deprecated Use {@link chainId} instead.
```
Comment (brad-decker) [inline index.d.ts:88]: this is daf.
